src/mpid/ch4/src/ch4r_request.h

```diff
@@ -147,6 +147,7 @@ static inline void MPIDI_anysrc_free_partner(MPIR_Request * rreq)
             /* NM, complete and free SHM partner */
             MPIDI_REQUEST_ANYSOURCE_PARTNER(rreq) = NULL;
             MPIDI_REQUEST_ANYSOURCE_PARTNER(anysrc_partner) = NULL;
+            /* FIXME: bug, only complete after setting status */
```

Comment:
It seems like the fix is simply to move the `MPID_Request_complete` down a line?

Reply:
I agree. I'll do that and remove the FIXME comment.